assets/tile-data.lisp

```diff
@@ -55,20 +55,39 @@
                      :sources (list (make-image-source pixel-data width height :unsigned-byte :rg-integer))
                      :tileset tileset)))
 
-(defun decode-tiled-tileset (data path asset)
-  (list :tileset (generate-resources 'image-loader (merge-pathnames (gethash "image" data) path)
-                                     :resource (resource asset (gethash "name" data))
-                                     :texture-class 'tileset
-                                     :width (gethash "imagewidth" data)
-                                     :height (gethash "imageheight" data)
-                                     :tile-size (vec (gethash "tileheight" data)
-                                                     (gethash "tilewidth" data)))
-        :first-id (gethash "firstgid" data)
-        :columns (gethash "columns" data)))
+(defun tileset-source (tileset)
+  "If the tileset is 'unembedded', yields the path to the .tsj/.json file where
+the actual tileset definition can be found. If the embedded tileset was saved in
+XML format, raise a condition.
+
+If the tileset is embedded, this yields NIL."
+  (let ((source (gethash "source" tileset)))
+    (when source
+      (let* ((as-path (pathname-utils:pathname* source))
+             (ext (pathname-type as-path)))
+        (if (or (string-equal "tsj" ext)
+                (string-equal "json" ext))
+            as-path
+            (error "Embedded tileset found, but not JSON format: ~a" as-path))))))
```

Comment:
Perhaps this function should be "pure" and simply yield what it found, and then the caller (`decode-tiled-tileset`) can be the one to raise the condition.